app/Insightly/Listeners/CreateProjectWithCoupon.php

```diff
@@ -25,6 +18,7 @@
 final class CreateProjectWithCoupon implements ShouldQueue
 {
     use Queueable;
+    use CreatesProject;
```

Comment:
My only (very minor) suggestion would be to make this trait a `CreateProject` class instead, maybe even invokable, which you instantiate directly in the constructor of `CreateProjectWithCoupon` and `CreateProjectWithOrganization` and run in their `handle()` methods. So basically the same but a class instead of a trait. But that way, the dependencies of the `CreateProject` class are clearer because it has its own properties that need to be injected (can be done via the `CreateProjectWithCoupon`/`CreateProjectWithOrganization` constructors of course)

Reply:
Very good remark. I also struggled with this design question during development. I will create an extra class `CreateProject` that has the required properties. I could even inject this class inside `CreateProjectWithCoupon` and `CreateProjectWithOrganization`.